packages/php/email-editor/src/Integrations/Core/Renderer/Blocks/class-media-text.php

```diff
@@ -0,0 +1,191 @@
+	 * @return string Media HTML content or empty string if not found.
+	 */
+	private function extract_media_from_html( string $block_content ): string {
+		// Extract media content (preserving any link wrapper).
+		$media_content = '';
+		if ( preg_match( '/<figure[^>]*class="[^"]*\bwp-block-media-text__media\b[^"]*"[^>]*>.*?<\/figure>/s', $block_content, $matches ) ) {
+			$media_content = $matches[0];
+		}
+
+		return $media_content;
+	}
+
+	/**
+	 * Build the email-friendly layout for media-text blocks.
+	 *
+	 * @param string            $media_content Media HTML content.
+	 * @param string            $text_content Text content.
+	 * @param array             $block_attrs Block attributes.
+	 * @param string            $block_content Original block content.
+	 * @param Rendering_Context $rendering_context Rendering context.
+	 * @return string Rendered HTML.
+	 */
+	private function build_email_layout( string $media_content, string $text_content, array $block_attrs, string $block_content, Rendering_Context $rendering_context ): string {
+		// Get original wrapper classes from block content.
+		$original_wrapper_classname = ( new Dom_Document_Helper( $block_content ) )->get_attribute_value_by_tag_name( 'div', 'class' ) ?? '';
+
+		// Get layout attributes.
+		$media_position     = $block_attrs['mediaPosition'] ?? 'left';
+		$vertical_alignment = $this->get_vertical_alignment_from_attributes( $block_attrs );
+		$media_width        = $this->get_media_width_from_attributes( $block_attrs );
+		$text_width         = 100 - $media_width; // Text takes the remaining width.
+
+		// Handle image linking for both "media" and "attachment" linkDestination types.
+		$link_destination = $block_attrs['linkDestination'] ?? '';
+		if ( in_array( $link_destination, array( 'media', 'attachment' ), true ) && ! empty( $block_attrs['href'] ) ) {
+			$media_content = $this->wrap_media_with_link( $media_content, $block_attrs['href'] );
+		}
+
+		// Get block styles using the Styles_Helper.
+		$block_styles = Styles_Helper::get_block_styles( $block_attrs, $rendering_context, array( 'padding', 'border', 'background', 'background-color', 'color' ) );
+		$block_styles = Styles_Helper::extend_block_styles(
+			$block_styles,
+			array(
+				'width'           => '100%',
+				'border-collapse' => 'collapse',
+				'text-align'      => 'left',
+			)
+		);
+
+		// Apply class and style attributes to the wrapper table.
+		$table_attrs = array(
+			'class' => 'email-block-media-text ' . $original_wrapper_classname,
+			'style' => $block_styles['css'],
+			'align' => 'left',
+			'width' => '100%',
+		);
+
+		// Build individual table cells.
+		$media_cell_attrs = array(
+			'style'  => sprintf( 'width: %d%%; padding: 10px; vertical-align: %s;', $media_width, $vertical_alignment ),
+			'valign' => $vertical_alignment,
+		);
+		$text_cell_attrs  = array(
+			'style'  => sprintf( 'width: %d%%; padding: 10px; vertical-align: %s;', $text_width, $vertical_alignment ),
```

Comment:
I noticed that the text block has [a hardcoded `padding: 0 8%;`](https://github.com/WordPress/gutenberg/blob/e9daf2a677553f6c29a81fe6bb07955722864a3c/packages/block-library/src/media-text/style.scss#L53). Not sure if it makes sense to apply the same for email rendering, but thought it was worth pointing out.

Reply:
That makes sense to me 👍 Updated in https://github.com/woocommerce/woocommerce/pull/60752/commits/2912574b896a74db09f22affc98b34b51c60b854